src/monsters.cpp

```diff
@@ -82,11 +82,72 @@ ConditionDamage* Monsters::getDamageCondition(ConditionType_t conditionType, int
 {
 	ConditionDamage* condition =
 	    static_cast<ConditionDamage*>(Condition::createCondition(CONDITIONID_COMBAT, conditionType, 0, 0));
-	condition->setParam(CONDITION_PARAM_TICKINTERVAL, tickInterval);
-	condition->setParam(CONDITION_PARAM_MINVALUE, minDamage);
-	condition->setParam(CONDITION_PARAM_MAXVALUE, maxDamage);
-	condition->setParam(CONDITION_PARAM_STARTVALUE, startDamage);
 	condition->setParam(CONDITION_PARAM_DELAYED, 1);
+
+	uint32_t defaultTickInterval = 2000;
+	uint32_t defaultTickInterval2 = 0;
+	ConditionDamageFormula_t damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+	int32_t rounds = 0;
+	int32_t damagePerRound = 0;
+
+	if (conditionType == CONDITION_FIRE) {
+		defaultTickInterval = 9000;
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		damagePerRound = 10;
+	} else if (conditionType == CONDITION_POISON) {
+		defaultTickInterval = 4000;
+		damageType = CONDITION_DAMAGE_FORMULA_LOG;
+	} else if (conditionType == CONDITION_ENERGY) {
+		defaultTickInterval = 11000;
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		damagePerRound = 25;
+	} else if (conditionType == CONDITION_DROWN) {
+		defaultTickInterval = 5000; // Need to find out tick rate
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		damagePerRound = 20;
+	} else if (conditionType == CONDITION_FREEZING) {
+		defaultTickInterval = 8000;
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		damagePerRound = 8;
+	} else if (conditionType == CONDITION_CURSED) {
+		defaultTickInterval = 4000;
+		damageType = CONDITION_DAMAGE_FORMULA_GROWTH;
+	} else if (conditionType == CONDITION_DAZZLED) {
+		defaultTickInterval = 10000;
+		defaultTickInterval2 = 12000;
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		damagePerRound = 20;
+	} else if (conditionType == CONDITION_BLEEDING) {
+		defaultTickInterval = 4000;
+		damageType = CONDITION_DAMAGE_FORMULA_CONSTANT;
+		// Not clear on how this works, it might be random damage per turn between 1 and 19?
+		damagePerRound = 15;
+	}
+	if (tickInterval == 0) {
+		tickInterval = defaultTickInterval;
+	} else {
+		defaultTickInterval2 = tickInterval;
+	}
+	if (startDamage > 0) {
+		damagePerRound = startDamage;
+	}
+
+	// Maybe better to just check if condition is dazzled? It would make more sense if we need special block for bleeding too
+	if ((damageType == CONDITION_DAMAGE_FORMULA_CONSTANT) && (defaultTickInterval2 != 0)) {
+		rounds = normal_random(minDamage, maxDamage) / damagePerRound;
+		for (int32_t i = 0; i <= rounds; i++) {
+			int32_t randTickInterval = uniform_random(tickInterval, defaultTickInterval2);
+			condition->addDamage(1, randTickInterval, damagePerRound * -1);
+		}
```

Comment:
Execute addDamage with rounds=0? HM

```suggestion
		for (uint32_t i = 0; i <= rounds; i++) {
			int32_t randTickInterval = uniform_random(tickInterval, defaultTickInterval2);
			condition->addDamage(1, randTickInterval, damagePerRound * -1);
		}
```

or
```suggestion
		do {
			int32_t randTickInterval = uniform_random(tickInterval, defaultTickInterval2);
			condition->addDamage(1, randTickInterval, damagePerRound * -1);
		} while (rounds-- != 0);
```